pkg/libcoap/Makefile

```diff
@@ -3,6 +3,9 @@ PKG_URL=https://github.com/obgm/libcoap
 PKG_VERSION=ef41ce5d02d64cec0751882ae8fd95f6c32bc018
 PKG_LICENSE=BSD-2-Clause
 
+# GCC 7.x fails on (intentional) fallthrough, thus disable implicit-fallthrough.
+CFLAGS += -Wno-implicit-fallthrough
```

Comment:
Shouldn't this rather go in `pkg/libcoap/Makefile.include`?

Reply:
mhm, @gebart was suggesting to use `Makefile` not `Makefile.include` so the option is only effective when compiling the package. IIRC putting it into `Makefile.include` exposes this to RIOT completely.

Reply:
Ah, you are right... But with `emb6` this doesn't work somehow.

Reply:
I'll look into it

Reply:
Ah, because `emb6` has submodules and `libcoap` and `oonf_api` do not... So at least for those submodules we somehow also need to activate `-Wno-implicit-fallthrough`... I think it is easier just to take the patch from smlng/RIOT#3 
for the moment, since porting to newest emb6 isn't that simple ;-).

Reply:
@smlng see https://github.com/RIOT-OS/RIOT/pull/8571